### PR TITLE
ldap: Add client certificate support

### LINKED ISSUE #4066: ldap_search: Client Certificate Support
### Summary

I am trying to get LDAP users from Google LDAP, but it requires client certificates. Due to lack of client certificate support, it's required to run stunnel as a middleware between Ansible and Google LDAP servers. It would be great to implement client certificate support to support direct connection to LDAP servers which require client certifiactes.

### Issue Type

Feature Idea

### Component Name

ldap_search

### Additional Information

_No response_

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by Gnonthgol**
##### SUMMARY
Add new arguments ``client_cert`` and ``client_key`` to the ldap modules in order to allow certificate authentication.

Fixes #4066

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
plugins/module_utils/ldap.py

##### ADDITIONAL INFORMATION
The functional part of the code is copied from work by @andrewshulgin with tests added by myself.

### CODE REVIEW

File: changelogs/fragments/6668-ldap-client-cert.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - ldap - add new arguments ``client_cert`` and ``client_key`` to the ldap modules in order to allow certificate authentication (https://github.com/ansible-collections/community.general/pull/6668).
```
Comment (felixfontein) [inline changelogs/fragments/6668-ldap-client-cert.yml:2]:
```suggestion
  - ldap_* - add new arguments ``client_cert`` and ``client_key`` to the LDAP modules in order to allow certificate authentication (https://github.com/ansible-collections/community.general/pull/6668).
```

File: plugins/doc_fragments/ldap.py [hunk 2]
```diff
@@ -29,6 +29,20 @@ class ModuleDocFragment(object):
       - Set the path to PEM file with CA certs.
     type: path
     version_added: "6.5.0"
+  client_cert:
+    type: path
+    description:
+      - PEM formatted certificate chain file to be used for SSL client authentication.
+      - Required if I(client_key) is defined.
+    aliases: [certfile]
```
Comment (felixfontein) [inline plugins/doc_fragments/ldap.py:37]: Are the aliases really needed? I would avoid adding aliases if possible.

File: plugins/doc_fragments/ldap.py [hunk 3]
```diff
@@ -29,6 +29,20 @@ class ModuleDocFragment(object):
       - Set the path to PEM file with CA certs.
     type: path
     version_added: "6.5.0"
+  client_cert:
+    type: path
+    description:
+      - PEM formatted certificate chain file to be used for SSL client authentication.
+      - Required if I(client_key) is defined.
```
Comment (felixfontein) [inline plugins/doc_fragments/ldap.py:36]:
```suggestion
      - Required if O(client_key) is defined.
```

File: plugins/doc_fragments/ldap.py [hunk 4]
```diff
@@ -29,6 +29,20 @@ class ModuleDocFragment(object):
...
+    aliases: [certfile]
+    version_added: "7.1.0"
+  client_key:
+    type: path
+    description:
+      - PEM formatted file that contains your private key to be used for SSL client authentication.
+      - Required if I(client_cert) is defined.
```
Comment (felixfontein) [inline plugins/doc_fragments/ldap.py:43]:
```suggestion
      - Required if O(client_cert) is defined.
```

File: plugins/doc_fragments/ldap.py [hunk 5]
```diff
@@ -29,6 +29,20 @@ class ModuleDocFragment(object):
...
+    aliases: [keyfile]
```
Comment (felixfontein) [inline plugins/doc_fragments/ldap.py:44]: Same here.

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

Please also modify the modules so that they pass `required_together=[('client_cert', 'client_key')]` to `AnsibleModule()`. That way the argument spec validation ensures that the user gets a nice error message if they only provide one of the two.

File: plugins/doc_fragments/ldap.py (hunk 2 above)
Comment (Gnonthgol) [reply plugins/doc_fragments/ldap.py:37]: They were in the code I copied. I do not see any reason to keep them.

**Review by felixfontein — APPROVED** [review]
Will merge if nobody objects in ~a week.

**Comment by felixfontein** [comment]
@Gnonthgol thanks for implementing this!

**Pull request merged**